back/lib/tasks/single_use/20250421_update_published_participants_data.rake

```diff
@@ -0,0 +1,117 @@
+namespace :single_use do
+  desc 'Update published participants data'
+  task update_published_participants_data: :environment do
+    def resolved_name(node)
+      type = node['type']
+      type['resolvedName'] if type.is_a?(Hash)
+    end
+
+    def floor_month(date_str)
+      throw ArgumentError, 'Invalid date format' unless date_str.length == 10
```

Comment:
I think `raise` is probably better than `throw` here.

`throw` Looking at the context of the `floor_month` method, I don't think there's a valid reason for using `throw` instead of `raise` here:

```ruby
def floor_month(date_str)
  throw ArgumentError, 'Invalid date format' unless date_str.length == 10
  "#{date_str.slice(0, 8)}01"
end
```

This appears to be a simple mistake rather than an intentional choice, because:

`throw` in Ruby is intended for non-local control flow with corresponding `catch` blocks.